clippy_utils/src/lib.rs

```diff
@@ -1648,6 +1648,18 @@ pub fn is_integer_literal(expr: &Expr<'_>, value: u128) -> bool {
     false
 }
 
+/// Checks whether the given expression is a constant integer of the given value
+/// as a floating point literal.
+pub fn is_floating_point_integer_literal(expr: &Expr<'_>, value: u32) -> bool {
```

Comment:
What about just accepting f64 directly and naming it "is_float_literal"? That feels like it could be more generally applicable possibly

Reply:
Yes. Done.